feat: October Changelog (#2883)
gitpod-io/website: src/lib/contents/changelog/2022-10-31.md

The standard pay-as-you-go rate for paid users, and for teams, is €/$ 0.036 per credit. Existing paid plans need to be cancelled for usage-based billing to work.

_Prices and terms are subject to change._

## Projects under personal accounts deprecated

Projects under personal accounts were deprecated with [PR #13414](https://github.com/gitpod-io/gitpod/pull/13414). Please add new projects under a team. Existing personal projects will continue to work for now, but may require migration to a team in future.

## Start New Workspace with this prebuild

![Prebuild Detail page - Start new (workspace with this prebuild) button](/images/changelog/2022-10-31/start-new-workspace-with-this-prebuild.jpg)

As a first step toward improving prebuilds for busy repos, we have shipped the ability to start workspaces using prebuilds built against older git commits. To see the button for this, click on the "Prebuild Ready" link in your project prebuild list.

Expect more improvements in this area soon. 🚀

## New ports view in VS Code Desktop

<!-- TODO: Add image -->

> Due to limitations with the [Remote - SSH extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh), we cannot extend the default ports view. To give you all the power of Gitpod, we've added a new tab, <kbd>EXPOSED PORTS</kbd>, which gives you all the power of the regular ports view of _Remote - SSH_, however with additional functionality like port visibility (public/private).

Reviewer: > which gives you all the power of the regular ports view of _Remote - SSH_, however with additional functionality like port visibility (public/private).

We should change. It does provide the same functionality as the default view. Ports view for forwarded ports, i.e. how access in local network. Exposed ports is about controlling access to ports on Internet.

- [New "exposed ports" view in VS Code Desktop 🔌](#new-exposed-ports-view-in-vs-code-desktop)
- [Ports sorting now based on your gitpod.yml definition ⏬](#ports-sorting-now-based-on-your-gitpodyml-definition)
- [Easily find information about your workspace with `gp info` ℹ️](#easily-find-information-about-your-workspace-with-gp-info)
- [VS Code ports view is now responsive ⏩](#vs-code-ports-view-is-now-responsive)
- [Projects now only created under teams, not personal accounts ⚠️](#projects-now-only-created-under-teams-not-personal-accounts)

## Even more power! &#x26A1; Try large workspaces with usage-based billing &#x1F389;

> **Important:** Prices and terms are subject to change.

With [usage-based billing](https://www.gitpod.io/docs/configure/billing/usage-based-billing) you get early access to [workspace classes](https://www.gitpod.io/docs/configure/workspaces/workspace-classes), allowing you to customize your workspace resources. To get access to [usage-based billing](https://www.gitpod.io/docs/configure/billing/usage-based-billing) for both teams and individual users, please [contact us](https://www.gitpod.io/contact/support).

> **How does usage-based billing work?**

Reviewer 2: It's unclear how usage-based billing works for teams.

- [Individual users](/docs/configure/billing/usage-based-billing#configure-personal-billing) receive 500 free credits per month on the free plan, these credits are equivalent to 50 Standard workspace hours or 25 Large workspace hours.

> - For users who upgrade to the personal paid plan at €/$ 9.00 per month, workspace usage allowance increases to 1000 credits.

Reviewer 2: ```suggestion
- Individual users who upgrade to the personal paid plan at €/$ 9.00 per month, workspace usage allowance increases to 1000 credits.
```

```

See the [Gitpod CLI](/docs/references/gitpod-cli#info) reference page for more.

> [Default environment variables](/docs/configure/projects/environment-variables#default-environment-variables) will soon be deprecated, please update all scripts to use `gp info --json`.

## VS Code ports view is now responsive &#x23E9;

<figure>
<video controls playsinline autoplay loop muted class="shadow-medium w-full rounded-xl max-w-3xl mt-x-small" alt="Responsive ports view in VS Code" src="/images/changelog/2022-10-31/responsive-ports.webm" type="video/webm"></video>
    <figcaption><i>New ports view on left navigation, with responsive design</i></figcaption>
</figure>

The ports view in VS Code is now responsive depending on the ports view width. The view defaults to the bottom VS Code navigation bar, however, the ports view can be drag-and-dropped to the sidebar if you prefer. You can access all port actions as normal on all views. In larger views, more information about your ports is shown.

<!-- TODO: Should this wording not be visibilty? -->

For feedback let us know on the [ports experience GitHub issue](https://github.com/gitpod-io/gitpod/issues/9262).

> ## Projects now only created under teams, not personal accounts ⚠️

Reviewer 2: Would it make sense to add a blurp on the reasoning?

Author: CC: @jldec

- [#13642](https://github.com/gitpod-io/gitpod/pull/13642) - Update IntelliJ IDEA IDE image to version 222.4345.14. <Contributors usernames="felladrin, filiptronicek" />
- [#13836](https://github.com/gitpod-io/gitpod/pull/13836) - Update GoLand IDE image to version 222.4345.24. <Contributors usernames="felladrin" />
- [#13747](https://github.com/gitpod-io/gitpod/pull/13747) - Fixed the auto-port-forwarding on JetBrains EAP IDEs <Contributors usernames="felladrin,andreafalzetti" />
- [#13797](https://github.com/gitpod-io/gitpod/pull/13797) - Update PyCharm IDE image to version 222.4345.23. <Contributors usernames="felladrin" />
- [#13758](https://github.com/gitpod-io/gitpod/pull/13758) - Update RubyMine IDE image to version 222.4345.14. <Contributors usernames="felladrin" />
- [#13757](https://github.com/gitpod-io/gitpod/pull/13757) - Update WebStorm IDE image to version 222.4345.14. <Contributors usernames="felladrin" />
- [#13759](https://github.com/gitpod-io/gitpod/pull/13759) - Update PhpStorm IDE image to version 222.4345.15. <Contributors usernames="felladrin" />

> ## Server

Reviewer 2: What's the difference between Server and Dashboard?

Reviewer 3: Merged <kbd>Server</kbd> into <kbd>Dashboard</kbd> in https://github.com/gitpod-io/website/pull/2883/commits/d1d6ea97d06f6eda268268fddd19ecbae118cb62, thanks for the feedback here, I really just went through component labels and made the config reflect that.

### VS Code Desktop

- [#26](https://github.com/gitpod-io/gitpod-vscode-desktop/pull/26) - Add `Install Local Extensions` command to explicitly install local extensions.
  On connection try to install local extensions but quietly without notifications. <Contributors usernames="akosyakov,mustard-mh" />

## Workspace

- [#13268](https://github.com/gitpod-io/gitpod/pull/13268) - Show an error if the workspace's persistent volume is smaller than the restore volume snapshot <Contributors usernames="jenting,utam0k" />
- [#13495](https://github.com/gitpod-io/gitpod/pull/13495) - All running processes receive a SIGTERM when a workspace shuts down. <Contributors usernames="svenefftinge,akosyakov,csweichel,easyCZ" />

### Workspace Images

- [#956](https://github.com/gitpod-io/workspace-images/pull/956) - Use tailscale 1.32 or later and avoid DNS issues <Contributors usernames="kylos101,aledbf,atduarte,mrzarquon,sagor999" />
- [#953](https://github.com/gitpod-io/workspace-images/pull/953) - Update `brew` layer to get recent versions of all available homebrew packages <Contributors usernames="andrew-farries,aledbf,sagor999" />

## Fixes and improvements

> - [#13768](https://github.com/gitpod-io/gitpod/pull/13768) - Add support for opening specific prebuilds <Contributors usernames="jankeromnes,csweichel,geropl,svenefftinge" />

Author: Unless we move Release Note (or override with manual change) I don't think there is a way of detecting that an item is already highlighted. That said, I'm actually in favour of leaving it here, for users who might happen to want to do a deep dive (we could discuss more how the automation / process should handle these cases).

> - [#13768](https://github.com/gitpod-io/gitpod/pull/13768) - Add support for opening specific prebuilds <Contributors usernames="jankeromnes,csweichel,geropl,svenefftinge" />

Reviewer 2: > Unless we move Release Note (or override with manual change) I don't think there is a way of detecting that an item is already highlighted

Yeah, I removed release notes of another PR (from workspace) for that reason.

- [#13822](https://github.com/gitpod-io/gitpod/pull/13822) - Support multi-line environment variables in SSH <Contributors usernames="iQQBot,filiptronicek,mustard-mh" />

> - [#13831](https://github.com/gitpod-io/gitpod/pull/13831) - Reliably close Workspaces which fail to start for whatever reason <Contributors usernames="geropl,easyCZ" />

Reviewer 2: @filiptronicek an example of something that is from WebApp but from a user POV imo it should be in the Workspace section. 

I don't know how to fix this, given that we rely on the team label 🤔 https://github.com/gitpod-io/website/blob/ft/auto-changelog-prs/scripts/changelog/lib/config.js

Reviewer 3: I think what we need is a component label for workspace, so that we can primarily rely on that and if it isn't present, fall back to the team label, @atduarte.

- [#13801](https://github.com/gitpod-io/gitpod/pull/13801) - Implement a 'Use Last Successful Prebuild' workspace creation mode <Contributors usernames="jankeromnes,geropl,mbrevoort" />
- [#13664](https://github.com/gitpod-io/gitpod/pull/13664) - Enable the protected secrets by default <Contributors usernames="kylos101,jenting,nandajavarma,svenefftinge,utam0k" />
- [#13745](https://github.com/gitpod-io/gitpod/pull/13745) - Fix branch context for BitBucket Server <Contributors usernames="AlexTugarev,easyCZ" />
- [#2683](https://github.com/gitpod-io/website/pull/2683) - Add Azure single-cluster guide <Contributors usernames="adrienthebo,MrSimonEmms,mrzarquon,utam0k" />
- [#2919](https://github.com/gitpod-io/website/pull/2919) - Add an example for installing docker in a custom workspace image <Contributors usernames="WVerlaek,atduarte,kylos101,utam0k" />
- [#2908](https://github.com/gitpod-io/website/pull/2908) - Instructs users to ensure they have tailscale 1.32 or later <Contributors usernames="mrzarquon,Siddhant-K-code,kylos101,sagor999" />

> - [#2886](https://github.com/gitpod-io/website/pull/2886) - New section on configuring a custom Dockerfile with non-gitpod base image <Contributors usernames="WVerlaek,axonasif,jenting,kylos101" />

Reviewer 2: Should be in the Docs section. Already has the label

Reviewer 3: Should be good now :)

<img width="909" alt="image" src="https://user-images.githubusercontent.com/29888641/199059905-fa70fd9f-a4c6-4e55-b3cd-a537158557d1.png">

> [Default environment variables](/docs/configure/projects/environment-variables#default-environment-variables) will soon be deprecated, please update all scripts to use `gp info --json`.

## VS Code ports view is now responsive &#x23E9;

<figure>
<video controls playsinline autoplay loop muted class="shadow-medium w-full rounded-xl max-w-3xl mt-x-small" alt="Responsive ports view in VS Code" src="/images/changelog/2022-10-31/responsive-ports.webm" type="video/webm"></video>
    <figcaption><i>New ports view on left navigation, with responsive design</i></figcaption>
</figure>

The ports view in VS Code is now responsive depending on the ports view width. The view defaults to the bottom VS Code navigation bar, however, the ports view can be drag-and-dropped to the sidebar if you prefer. You can access all port actions as normal on all views. In larger views, more information about your ports is shown.

<!-- TODO: Should this wording not be visibilty? -->

For feedback let us know on the [ports experience GitHub issue](https://github.com/gitpod-io/gitpod/issues/9262).

## Projects now only created under teams, not personal accounts ⚠️

> - You can no longer create Gitpod Projects under personal accounts.

Author: **Question:** Does this restriction only apply to new accounts?

Reviewer 4: - existing accounts with no personal projects cannot create new ones because the personal account link at the top is gone. 
- accounts with personal projects still have access to them with a personal account "pseudo-team", and currently they can add more - but we show a [very scary warning](https://github.com/gitpod-io/gitpod/issues/13892) at the top.

- All projects must now be created as part of a team.
- Existing personal projects will continue to work.
<!-- TODO: Is there an option to migrate projects? -->

<!--- BEGIN_AUTOGENERATED_CHANGES -->

## Dashboard

> - [#13538](https://github.com/gitpod-io/gitpod/pull/13538) - Update free plan name <Contributors usernames="gtsiolis,andrew-farries,filiptronicek" />

Author: **Suggestion:** Could this be "Open Source" plan now renamed to "Free" plan to be more descriptive? 

CC: @gtsiolis 

**Note:** Update PR "Release Notes" so we can re-run the script to show changes here.

Reviewer 4: I think we were already using Free on the website - we just fixed the dashboard to make it consistent :)

Reviewer 5: Came here to confirm what @jldec said, see https://github.com/gitpod-io/gitpod/issues/13229.

**How does usage-based billing work?**

- [Individual users](/docs/configure/billing/usage-based-billing#configure-personal-billing) receive 500 free credits per month on the free plan, these credits are equivalent to 50 Standard workspace hours or 25 Large workspace hours.
- For users who upgrade to the personal paid plan at €/$ 9.00 per month, workspace usage allowance increases to 1000 credits.
- The standard pay-as-you-go rate for paid users, and [for teams](/docs/configure/billing/usage-based-billing#configure-team-billing), is €/$ 0.036 per credit.
- Existing paid plans need to be cancelled for usage-based billing to work.

<br/>

<LinkButton href="/contact/support" size="medium" variant="primary">I want early access!</LinkButton>

## Start workspaces based on existing prebuilds ⚡️

![Prebuild Detail page - Start new (workspace with this prebuild) button](/images/changelog/2022-10-31/start-new-workspace-with-this-prebuild.jpg)

> _Caption: Start a new workspace based on an existing prebuild_

Reviewer 6: Do we need the word `caption` here? The other images do not seem to have it a caption. The video that shows the VS code ports view does have a caption, but that one is done via html: `<figcaption><i>New ports view on left navigation, with responsive design</i></figcaption>`

## VS Code ports view is now responsive &#x23E9;

<figure>
<video controls playsinline autoplay loop muted class="shadow-medium w-full rounded-xl max-w-3xl mt-x-small" alt="Responsive ports view in VS Code" src="/images/changelog/2022-10-31/responsive-ports.webm" type="video/webm"></video>
    <figcaption><i>New ports view on left navigation, with responsive design</i></figcaption>
</figure>

The ports view in VS Code is now responsive depending on the ports view width. The view defaults to the bottom VS Code navigation bar, however, the ports view can be drag-and-dropped to the sidebar if you prefer. You can access all port actions as normal on all views. In larger views, more information about your ports is shown.

<!-- TODO: Should this wording not be visibilty? -->

For feedback let us know on the [ports experience GitHub issue](https://github.com/gitpod-io/gitpod/issues/9262).

## Projects now only created under teams, not personal accounts ⚠️

- You can no longer create Gitpod Projects under personal accounts.
- All projects must now be created as part of a team.
- Existing personal projects will continue to work.

## Using private workspace images (Alpha)

> Using private Docker images as workspace images has been possible for a while but it stayed mostly under the radar.
> 
> We have updated documentation with the details, so that you can make the most of it. See [Use a private Docker image](https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-private-docker-image) for more.

Author: **Suggestion**
* Swap language to be based on the user e.g. "_You can now do X_", as opposed to "_We did X_"
* Add call to action on where to report issues (since it's marked as alpha)
* Remove "under the radar" wording (could be ambiguous)

```suggestion
Did you know that you can [use a private Docker image](https://www.gitpod.io/docs/configure/workspaces/workspace-image#use-a-private-docker-image) with Gitpod? Private Docker image support for workspace images has been possible for some time, however it was an undocumented feature. We've now updated our documentation so that you can more easily give try out the feature. As the feature is currently alpha, please [raise an issue](https://github.com/gitpod-io/gitpod/issues/new/choose) for any feedback. 
```

---
title: October Gitpod Release 2022
excerpt: The latest Gitpod updates for October 2022, including new JetBrains IDEs, private workspace images and lots of other fixes and improvements.
date: 2022-10-31
image: 2022-10-31.jpg
ogImage: 2022-10-31-og.jpg
alt:
---

<script>
  import Contributors from "$lib/components/changelog/contributors.svelte";
  import LinkButton from "$lib/components/ui-library/link-button/link-button.svelte"
</script>

> Welcome to the October edition of the Gitpod Changelog!

Reviewer 7: ```suggestion
Welcome to the October 2022 edition of the Gitpod Changelog!
```

- For users who upgrade to the personal paid plan at €/$ 9.00 per month, workspace usage allowance increases to 1000 credits.
- The standard pay-as-you-go rate for paid users, and [for teams](/docs/configure/billing/usage-based-billing#configure-team-billing), is €/$ 0.036 per credit.
- Existing paid plans need to be cancelled for usage-based billing to work.

<br/>

<LinkButton href="/contact/support" size="medium" variant="primary">I want early access!</LinkButton>

## Start workspaces based on existing prebuilds ⚡️

![Prebuild Detail page - Start new (workspace with this prebuild) button](/images/changelog/2022-10-31/start-new-workspace-with-this-prebuild.jpg)
_Start a new workspace based on an existing prebuild_

You can now start workspaces using existing prebuilds. This helps when you have a repo with many commits, so you're not left waiting for the latest prebuild to finish. To get started, select a prebuild from your project prebuild list, and click "**New Workspace (with this prebuild)**"

> ## RubyMine and Webstorm IDEs now in beta &#x1F680;

Reviewer 7: ```suggestion
## JetBrains RubyMine and WebStorm IDEs now in beta &#x1F680;
```

> ![](/images/blog/webstorm-rubymine-now-in-beta/webstorm-rubymine-announcement-gitpod.webp)

Reviewer 7: ```suggestion
![JetBrains RubyMine and WebStorm logos connected to a Gitpod logo](/images/blog/webstorm-rubymine-now-in-beta/webstorm-rubymine-announcement-gitpod.webp)
```

> We are continuing to expand support for JetBrains IDEs as part of our [official partnership](https://www.gitpod.io/blog/gitpod-jetbrains) with [Jetbrains](https://www.jetbrains.com/).

Reviewer 7: ```suggestion
We are continuing to expand support for JetBrains IDEs as part of our [official partnership](https://www.gitpod.io/blog/gitpod-jetbrains) with [JetBrains](https://www.jetbrains.com/).
```

See the [RubyMine & WebStorm announcement](https://www.gitpod.io/blog/webstorm-rubymine-now-in-beta) for more.

## New "exposed ports" view in VS Code Desktop &#x1F50C;

> ![](/images/editors/vscode-exposed-ports.png)

Reviewer 7: ```suggestion
![A panel in VS Code Desktop listing open ports in the workspace](/images/editors/vscode-exposed-ports.png)
```

To give you all the power of Gitpod, we've added a new tab, <kbd>EXPOSED PORTS</kbd>, which gives you additional functionality in addition to the regular ports view of _Remote - SSH_, like port visibility (public/private).

> **Note:** Due to constraints with [Remote - SSH extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh), we cannot extend the default VS Code ports view.

## Ports sorting now based on your gitpod.yml definition &#x23EC;

Ports shown in all Gitpod clients are now ordered based on:

1. The order of the ports definition in the `.gitpod.yml`
2. All other ports are numerically sorted in ascending order.

> This change applies to all of VS Code, JetBrains, SSH and Gitpod CLI.

Reviewer 7: ```suggestion
This change applies to all of VS Code, JetBrains IDEs, SSH and the Gitpod CLI.
```

## VS Code ports view is now responsive &#x23E9;

<figure>
<video controls playsinline autoplay loop muted class="shadow-medium w-full rounded-xl max-w-3xl mt-x-small" alt="Responsive ports view in VS Code" src="/images/changelog/2022-10-31/responsive-ports.webm" type="video/webm"></video>
    <figcaption><i>New ports view on left navigation, with responsive design</i></figcaption>
</figure>

The ports view in VS Code is now responsive depending on the ports view width. The view defaults to the bottom VS Code navigation bar, however, the ports view can be drag-and-dropped to the sidebar if you prefer. You can access all port actions as normal on all views. In larger views, more information about your ports is shown.

<!-- TODO: Should this wording not be visibilty? -->

For feedback let us know on the [ports experience GitHub issue](https://github.com/gitpod-io/gitpod/issues/9262).

## Projects now only created under teams, not personal accounts ⚠️

- You can no longer create Gitpod Projects under personal accounts.
- All projects must now be created as part of a team.
- Existing personal projects will continue to work.

> ## Using private workspace images (Alpha)

Reviewer 7: ```suggestion
## Using private workspace images is now in alpha
```

---
title: October Gitpod Release 2022

> excerpt:

Reviewer 8: We should add an excerpt before merging

Reviewer 8: I added one in a newer commit. Feel free to adjust it 👍

> ---

Author: @filiptronicek - looks like GitHub Actions bot image is broken, can we filter bots from the contributor list? 👀 

![image](https://user-images.githubusercontent.com/5528307/197542678-0c1dee4c-5a7d-489e-8b53-4d422b16dc80.png)

Reviewer 3: Done! They shouldn't appear anymore.

- [#14054](https://github.com/gitpod-io/gitpod/pull/14054) - Update balance used progress indicator height <Contributors usernames="gtsiolis,filiptronicek,jankeromnes" />
- [#13414](https://github.com/gitpod-io/gitpod/pull/13414) - Deprecation of the ability to create projects under an individual account for new users. <Contributors usernames="svenefftinge,easyCZ,gtsiolis,jldec" />
- [#13621](https://github.com/gitpod-io/gitpod/pull/13621) - Fix rendering Personal/Team billing menu entries <Contributors usernames="geropl,laushinka" />

> - [#14081](https://github.com/gitpod-io/gitpod/pull/14081) – Add message: Prebuilds have been paused <Contributors usernames="AlexTugarev,gtsiolis,geropl,jankeromnes" />

Author: @filiptronicek - This was added manually, on the next run of the automation script, can we make sure this one was / is included? CC: @AlexTugarev

Reviewer 3: I wonder if we should change the release note on the PR itself to reflect what is the message we want to show in the changelog, @AlexTugarev?

Author: **Thought:** This is making me realise we need to make it more clear / obvious how these PR's end up in the changelog. The slackbot proposal could help, but also some docs, possibly in the template, too. 

[Link to internal message](https://gitpod.slack.com/archives/C01KGM9AW4W/p1666952585398059).

Reviewer 5: > This is making me realise we need to make it more clear / obvious how these PR's end up in the changelog.

Came here to upvote this.
